app/assets/javascripts/modules/single-file-upload.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+moj.Modules.singleFileUpload = {
+  input_class: '.single-file-upload',
+  spinnerImagePath: '/assets/spinner.gif',
+
+  init: function() {
+    var self = this;
+
+    if($(self.input_class).length) {
+      self.bindEvents();
+      self.preloadSpinner();
+    }
+  },
+
+  bindEvents: function() {
+    var self = this,
+        $el = $(self.input_class).eq(0),
+        $form = $el.closest('form');
+
+    $el.on('change', function() {
+      var statusText = '';
+
+      if($el.val()) {
+        statusText = 'SELECTED';
```

Comment:
This feels like something we should probably put into i18n...

But considering how much of a pain it was to get values out of Rails inside assets with the JS timeout (yay caching), that might not be worth the hassle. Thoughts @zheileman @tatyree @digitalronin?